src/index.ts

```diff
@@ -44,16 +44,61 @@ v1.post('/session/create', async (req, res) => {
 	if (authRes.status === 403)
 		return res.status(403).send('Invalid credentials');
 
-	const hashedServerId = await bcrypt.hash(serverId, 10);
+	const hashedServerId = crypto
+		.createHash('sha256')
+		.update(serverId)
+		.digest('hex')
+		.toString();
 
 	const session = new Session();
 
 	session.hashed_uuid = hashedServerId;
+	session.start = new Date();
+	session.end = new Date(Date.now() + EXPIRE_TIME_SECS * 1000);
 
-	const newSession = await AppDataSource.manager.save(session);
+	await AppDataSource.manager.save(session);
+
+	const token = jwt.sign({ hashedServerId }, JWT_SECRET, {
+		expiresIn: EXPIRE_TIME_SECS,
+	});
+
+	res.send(token);
+});
+
+v1.post('/session/refresh', async (req, res) => {
+	const decoded = verifyToken(req.body.token, true);
+	if (typeof decoded === 'string') return res.status(403).send(decoded);
+
+	const authRes = await fetch(AUTH_SERVER, {
+		method: 'POST',
+		body: JSON.stringify(req.body),
+	});
+
+	if (authRes.status === 403)
+		return res.status(403).send('Invalid credentials');
+
+	const hashedServerId = crypto
+		.createHash('sha256')
+		.update(req.body.serverId)
+		.digest('hex')
+		.toString();
+
+	if (hashedServerId !== req.body.hashedServerId)
+		return res.status(403).send('UUIDs do not match');
+
+	const sessionRepository = AppDataSource.getRepository(Session);
+
+	const session = await findLatestSessionByUUID(
+		decoded.hashedServerId,
+		sessionRepository,
+	);
+
+	session.end = new Date(session.end.getTime() + EXPIRE_TIME_SECS * 1000);
```

Comment:
This shouldnt add to the end but rather to the current time (Date.now())

Reply:
Initially I thought the same  but wouldn't that mean that the end gets changed because of delays?

Reply:
Wdym gets changed? Its supposed to get changed, if you add 5 minutes everytime someone makes a request to the refresh endpoint people can just make requests every second and end up with a token that doesnt expire in hours

Reply:
True I didnt even think about that

Reply:
Although the expiration time on the token is different to the session end value but it probably should still be changed

Reply:
It shouldnt be different, if we return a token with an expiry time of 5 minutes then that has the same expiry time as the session